dll/win32/browseui/regtreeoptions.cpp

```diff
@@ -1,59 +1,565 @@
- * License as published by the Free Software Foundation; either
- * version 2.1 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ * PROJECT:     browseui
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Common registry based settings editor
+ * COPYRIGHT:   Copyright 2024 Whindmar Saksit <whindsaks@proton.me>
  */
 
 #include "precomp.h"
 
-CRegTreeOptions::CRegTreeOptions()
+#define HKEY_GRAYED HKEY_REGTREEOPTION_GRAYED
+#define MAXVALUEDATA 100 // Max size for CheckedValue/UncheckedValue/DefaultValue
+
+enum {
+    TYPE_CHECK = 0,
+    TYPE_RADIO,
+    TYPE_GROUP,
+    TYPE_INVALID,
+    STATEIMAGESPERTYPE = 4,
+    STATE_CHECKOFF = TYPE_CHECK * STATEIMAGESPERTYPE,
+    STATE_CHECKON,
+    STATE_RADIOOFF = TYPE_RADIO * STATEIMAGESPERTYPE,
+    STATE_RADIOON,
+    STATE_GROUP    = TYPE_GROUP * STATEIMAGESPERTYPE,
+};
+
+C_ASSERT((STATE_CHECKON & 1) && (STATE_RADIOON & 1) && STATE_RADIOON < STATE_GROUP);
+
+static HBITMAP CreatDibBitmap(HDC hDC, UINT cx, UINT cy, UINT bpp)
 {
+    BITMAPINFO bi;
+    LPVOID pvBits;
+
+    ZeroMemory(&bi, sizeof(bi));
+    bi.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    bi.bmiHeader.biWidth = cx;
+    bi.bmiHeader.biHeight = cy;
+    bi.bmiHeader.biPlanes = 1;
+    bi.bmiHeader.biBitCount = bpp;
+    bi.bmiHeader.biCompression = BI_RGB;
+    return CreateDIBSection(hDC, &bi, DIB_RGB_COLORS, &pvBits, NULL, 0);
 }
 
-CRegTreeOptions::~CRegTreeOptions()
+static HIMAGELIST CreateStateImages(UINT ImageSize)
 {
+    enum { bpp = 32 };
+    UINT Failed = FALSE, State;
+    HIMAGELIST hIL = ImageList_Create(ImageSize, ImageSize, ILC_MASK | bpp, 4, 0);
+    if (!hIL)
+        return hIL;
+
+    INT BorderSize = max(1, min(ImageSize / 16, 4));
+    RECT Rect;
+    HDC hDC = CreateCompatibleDC(NULL);
+
+    HBITMAP hbmMask = CreateBitmap(ImageSize, ImageSize, 1, 1, NULL);
+    Failed |= hbmMask == NULL;
+
+    HBITMAP hbmData = CreatDibBitmap(hDC, ImageSize, ImageSize, bpp);
+    Failed |= hbmData == NULL;
+
+    const HGDIOBJ hbmOld = SelectObject(hDC, hbmMask);
+    Failed |= hbmOld == NULL;
+    SetRect(&Rect, 0, 0, ImageSize, ImageSize);
+    FillRect(hDC, &Rect, HBRUSH(GetStockObject(WHITE_BRUSH)));
+    InflateRect(&Rect, -BorderSize, -BorderSize);
+    FillRect(hDC, &Rect, HBRUSH(GetStockObject(BLACK_BRUSH)));
+
+    SelectObject(hDC, hbmData);
+    State = DFCS_BUTTONCHECK | DFCS_FLAT | DFCS_MONO;
+    DrawFrameControl(hDC, &Rect, DFC_BUTTON, State | 000000000000 | 0000000000000);
```

Comment:
No value in adding these things.
```suggestion
    DrawFrameControl(hDC, &Rect, DFC_BUTTON, State);
```